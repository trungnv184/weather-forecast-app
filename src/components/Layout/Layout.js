import React from "react";
import Header from "../Header/Header";

const Layout = ({children}) => {
  return (
    <React.Fragment>
      <Header />
      <main>{children}</main>
    </React.Fragment>
  );
};

export default Layout;
