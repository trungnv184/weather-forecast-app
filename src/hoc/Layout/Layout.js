import React from "react";
import Aux from "../Aux/Aux";
import Header from "../../components/Header/Header";

const Layout = (prop) => {
  const x = "122";
  return (
    <Aux>
      <Header />
      <main>{prop.children}</main>
    </Aux>
  );
};
export default Layout;
