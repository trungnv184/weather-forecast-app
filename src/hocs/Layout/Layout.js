import React from "react";
import Aux from "../Aux/Aux";
import Header from "../../components/Header/Header";
import PropTypes from "prop-types";

const Layout = ({children}) => {
  return (
    <Aux>
      <Header />
      <main>{children}</main>
    </Aux>
  );
};

Layout.propTypes = {
  children: PropTypes.node
};
export default Layout;
