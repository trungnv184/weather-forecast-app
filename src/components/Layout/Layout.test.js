import React from "react";
import {cleanup} from "@testing-library/react";
import {screen} from "@testing-library/dom";
import {renderComponentWithLocationProvider} from "../../test/testUtils";
import Layout from "./Layout";

beforeEach(() => {
  setupTest();
});
afterEach(cleanup);

test("should have default location when initalize component", () => {
  expect(screen.getByDisplayValue("Ho Chi Minh City")).toBeInTheDocument();
});

test("snapshot", () => {
  const {container} = setupTest();
  expect(container.firstChild).toMatchSnapshot();
});

const setupTest = (location) => {
  return renderComponentWithLocationProvider(<Layout />, {
    location: location,
    setLocation: () => {}
  });
};
