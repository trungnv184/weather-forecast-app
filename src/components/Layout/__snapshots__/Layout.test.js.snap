// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<header
  class="header bg-white border-bottom shadow-sm"
>
  <nav
    class="navbar navbar-expand-lg navbar-light"
  >
    <form
      class="form-inline"
    >
      <input
        class="mr-sm-2 form-control"
        placeholder="Search"
        type="text"
        value="Ho Chi Minh City"
      />
      <button
        class="btn btn-outline-success"
        type="button"
      >
        Search
      </button>
    </form>
  </nav>
</header>
`;
